### PR TITLE
(Feature): Implemented alternate logout messages using Redux

### PR DESCRIPTION
**Pull request by leoric-crown**
TODO: We need to check for valid token when switching components using Nav

### CODE REVIEW

File: src/utils/misc.js [hunk 1]
```diff
@@ -14,3 +16,22 @@ export const verifyToken = (token) => {
     })
   })
 }
+
+export const checkToken = (token, dispatch, history) => {
+  verifyToken(token)
+      .then((authedUser) => {
+        if (authedUser) {
+          dispatch(setAuthedUser(authedUser.email, "http://s3.amazonaws.com/37assets/svn/765-default-avatar.png", authedUser.isDM, authedUser._id))
```
Comment (leoric-crown) [inline src/utils/misc.js]: I think we might want to return true / false here instead and dispatch actions directly from Components.

Thoughts?

Comment (leoric-crown) [reply src/utils/misc.js]: The idea is that this method will be run when mounting all protected components and if necessary redirect the User to the Login component

File: src/components/Login.js [hunk 2]
```diff
@@ -7,48 +7,37 @@ import config from '../config.json';
...
-          })
-        } else {
-          localStorage.removeItem('DNDTOKEN')
-          this.props.history.push({
-            pathname: '/'
-          })
-        }
-      })
-    }
+    if (token) checkToken(token, this.props.dispatch, this.props.history)
```
Comment (leoric-crown) [inline src/components/Login.js]: This refactor (which we might want to move elsewhere) makes Components much leaner and easier to read

File: src/components/Characters.js [hunk 3]
```diff
@@ -26,35 +25,15 @@ class Characters extends React.Component {
...
-          verifyToken(token)
-          .then((authedUser) => {
-            if (authedUser) {
-              this.props.dispatch(setAuthedUser(authedUser.email, "http://s3.amazonaws.com/37assets/svn/765-default-avatar.png", authedUser.isDM, authedUser._id))
-              this.props.dispatch(handleInitialData(authedUser._id, token))
-            } else {
-              this.sendToLogin(true);
-            }
-          })
+          checkToken(token, this.props.dispatch, this.props.history)
```
Comment (leoric-crown) [inline src/components/Characters.js]: This refactor (which we might want to move elsewhere) makes Components much leaner and easier to read

File: src/utils/misc.js (hunk 1 above)
Comment (mquijanogv) [reply src/utils/misc.js]: Actually, been thinking of this. The correct approach imo, is to use the `verifyToken` api call as middleware in an action creator. so we could dispatch an action to an async action creator.

File: src/components/Login.js (hunk 2 above)
Comment (mquijanogv) [reply src/components/Login.js]: This is alright, but isn't  kind of hacky of sending dispatch and history to a function in utils? Dunno tbh

File: src/utils/misc.js (hunk 1 above)
Comment (mquijanogv) [reply src/utils/misc.js]: But still, removing unecessary code from components is a good thing. So :100:

File: src/components/Login.js (hunk 2 above)
Comment (mquijanogv) [reply src/components/Login.js]: But still, removing unecessary code from components is a good thing. So :100:

**Pull request merged**